proposals/0010-validating-resource-container-elements.md

```diff
@@ -0,0 +1,142 @@
+* Proposal: [0010](0010-validating-resource-container-elements.md)

```

Comment:
FWIW: I prefer the title of the PR "Resource element type validation" to the filename "validating resource container elements"

Reply:
Also, I think that this proposal is only about element types for RWBuffer?  Is it going to cover rules for any other resource types?  If not, then the title should probably reflect the limited scope.

Reply:
> Also, I think that this proposal is only about element types for RWBuffer? Is it going to cover rules for any other resource types? If not, then the title should probably reflect the limited scope.

I intend for this proposal to cover rules for all resource element types used in any resource context.

Reply:
> I intend for this proposal to cover rules for all resource element types used in any resource context.

In this PR?  Currently it only seems to cover buffers and the word "texture" doesn't appear in the document at all.  If the rules for buffers are the same for textures then that's fine, but it'd be good to call that out.